splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/DDLSingleTableConstantOperation.java

```diff
@@ -135,6 +135,21 @@ void dropConstraint(ConstraintDescriptor consDesc,TableDescriptor skipCreate, Li
             String schemaName = td.getSchemaName();
             dropIndex(td, indexConglom, (SpliceTransactionManager)lcc.getTransactionExecute(),lcc, schemaName, indexName);
         }
+
+        if (skipCreate != null && consDesc.getConstraintType() == DataDictionary.PRIMARYKEY_CONSTRAINT) {
+            ConglomerateDescriptorList cdl = skipCreate.getConglomerateDescriptorList();
+            DataDictionary dd = lcc.getDataDictionary();
+            TransactionController tc = lcc.getTransactionExecute();
+            for (ConglomerateDescriptor cd : cdl) {
+                if (cd.isPrimaryKey()) {
+                    cd.setIndexRowGenerator(new IndexRowGenerator());
```

Comment:
Is it necessary to set an empty `IndexRowGenerator` instead of just `null`?

Reply:
it can be null